lib/AST/Decl.cpp

```diff
@@ -3612,6 +3612,10 @@ void VarDecl::emitLetToVarNoteIfSimple(DeclContext *UseDC) const {
     // If the problematic decl is 'self', then we might be trying to mutate
     // a property in a non-mutating method.
     auto FD = dyn_cast_or_null<FuncDecl>(UseDC->getInnermostMethodContext());
+
+    // If this is a subscript getter, don't suggest adding mutating.
+    if (FD->getAccessorStorageDecl()->getGetter() == FD) return;
```

Comment:
I think the correct check would be `accessor->getAccessorKeywordLoc().isValid()`.

Multiple fix-its attached to the same diagnostic will be applied together. They must not overlap but other than that there should be no issues.

For the record, `isImplicit` implies that the entire thing is implicit, and since the body is written explicitly it wouldn't be appropriate here.

Stepping back, I'm honestly happy to not even *have* the fix-it for getters. Most people don't *really* want a mutating getter, nor would they want to think about the implications.